app/src/main/java/com/zulip/android/util/RemoveViewsOnScroll.java

```diff
@@ -43,25 +52,30 @@ public boolean onStartNestedScroll(CoordinatorLayout coordinatorLayout, View chi
     @SuppressLint("NewApi")
     @Override
     public void onNestedPreScroll(CoordinatorLayout coordinatorLayout, View child, View target, int dx, int dy, int[] consumed) {
-        if (dy > 0 && changeInYDir < 0 || dy < 0 && changeInYDir > 0) {
-            child.animate().cancel();
-            changeInYDir = 0;
-        }
+        try {
+            if (linearLayoutManager.findLastCompletelyVisibleItemPosition() < adapter.getItemCount() - 3) {
+                if (dy > 0 && changeInYDir < 0 || dy < 0 && changeInYDir > 0) {
+                    child.animate().cancel();
+                    changeInYDir = 0;
+                }
 
-        changeInYDir += dy;
-        if (changeInYDir > toolbarHeight && child.getVisibility() == View.VISIBLE && !isViewHidden)
-            hideView(child);
-        else if (changeInYDir < 0 && child.getVisibility() == View.GONE && !mIsShowing) {
-            if (child instanceof FloatingActionButton) {
-                if (chatBox == null)
-                    chatBox = coordinatorLayout.findViewById(R.id.messageBoxContainer);
-                if (chatBox.getVisibility() == View.VISIBLE) {
-                    return;
+                changeInYDir += dy;
+                if (changeInYDir > toolbarHeight && child.getVisibility() == View.VISIBLE && !isViewHidden)
+                    hideView(child);
+                else if (changeInYDir < 0 && child.getVisibility() == View.GONE && !mIsShowing) {
+                    if (child instanceof FloatingActionButton) {
+                        if (chatBox == null)
+                            chatBox = coordinatorLayout.findViewById(R.id.messageBoxContainer);
+                        if (chatBox.getVisibility() == View.VISIBLE) {
+                            return;
+                        }
+                    }
+                    showView(child);
                 }
             }
-            showView(child);
+        } catch (NullPointerException e) {
+            ZLog.logException(e);
```

Comment:
@vishwesh3 since these changes were suggested for the first commit, can you please make these changes in 4e426f81b6b5ebea0e12d40c97fcf0ad6e4d6454